src/backtracking/permutations.rs

```diff
@@ -1,63 +1,137 @@
-/*
-The permutations problem involves finding all possible permutations
-of a given collection of distinct integers. For instance, given [1, 2, 3],
-the goal is to generate permutations like
- [1, 2, 3], [1, 3, 2], [2, 1, 3], [2, 3, 1], [3, 1, 2], and [3, 2, 1].
- This implementation uses a backtracking algorithm to generate all possible permutations.
-*/
-
-pub fn permute(nums: Vec<i32>) -> Vec<Vec<i32>> {
-    let mut result = Vec::new(); // Vector to store the resulting permutations
-    let mut current_permutation = Vec::new(); // Vector to store the current permutation being constructed
-    let mut used = vec![false; nums.len()]; // A boolean array to keep track of used elements
-
-    backtrack(&nums, &mut current_permutation, &mut used, &mut result); // Call the backtracking function
-
-    result // Return the list of permutations
+//! This module provides a function to generate all possible distinct permutations
+//! of a given collection of integers using a backtracking algorithm.
+
+/// Generates all possible distinct permutations of a given vector of integers.
+///
+/// # Arguments
+///
+/// * `nums` - A vector of integers.
```

Comment:
Please mention that this function sorts `nums`.

Reply:
Will update the doc-string.